### PR TITLE
 Feat/practice session - start/stop control API and displayhint

### PR DESCRIPTION
**Pull request by mickelr**
# COMPLETES #< INSERT LINK TO ISSUE >
https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-575353

## This pull request addresses

- Practice session start/stop control API and related display hint
- Add new parameter "roles" for invite API

## by making the following changes

< DESCRIBE YOUR CHANGES >

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [x] Internal code refactor

## The following scenarios were tested

- Host/cohost can start and stop practice session  according to related display hints.
- Panelist/Attendee receive practice session on/off status.

### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [ ] I discussed changes with code owners prior to submitting this pull request

- [x] I have not skipped any automated checks
- [x] All existing and new tests passed
- [ ] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

## Release Notes

- **New Features**
	- Introduced practice session management capabilities, allowing users to start and stop practice sessions during meetings and webinars.
	- Added new display hints for practice sessions to enhance user interface interactions.
	- Enhanced role management logic for better control over practice session permissions.
	- Updated invite functionality to include role specifications for invited members.
	- Improved data structure for adding members by including role information.

- **Bug Fixes**
	- Improved error handling for media connections and practice session transitions.

- **Tests**
	- Expanded test coverage for the new practice session features, ensuring robust validation of functionality and error handling.
	- Added tests for the `getAddMemberBody` method to validate various input scenarios.

### CODE REVIEW

**Comment by Shreyas281299** [comment]
@coderabbit review

**Review by Shreyas281299 — CHANGES_REQUESTED** [review]
Please add the manual tests performed in the PR description.

**Comment by mickelr** [comment]
> Please add the manual tests performed in the PR description.

Done

File: packages/@webex/plugin-meetings/src/webinar/index.ts [hunk 1]
```diff
@@ -59,18 +61,48 @@ const Webinar = WebexPlugin.extend({
...
 
     return {isPromoted, isDemoted};
   },
+
+  /**
+   * start or stop practice session for webinar
+   * @param {boolean} enabled
+   * @returns {Promise}
+   */
+  setPracticeSessionState(enabled) {
```
Comment (Shreyas281299) [inline packages/@webex/plugin-meetings/src/webinar/index.ts:83]: Question: can you point me to where we are using this method, I was unable to find it

**Review by Shreyas281299 — APPROVED** [review]
Looks good to me, please see a query that I had.

**Review by JudyZhuHz — APPROVED** [review]
LGTM

**Pull request merged**